site/components/pages/Pledge/lib/firebase.ts

```diff
@@ -53,21 +58,99 @@ export const findOrCreatePledge = async (
     const currentPledge = pledgeRef.data();
 
     if (!currentPledge) return null;
-
-    verifySignature({
-      address: currentPledge.ownerAddress,
-      signature: params.signature,
-      nonce: currentPledge.nonce.toString(),
-    });
-
-    const pledgeAttributes = putPledgeAttributes({
-      currentPledgeValues: currentPledge,
-      newPledgeValues: params.pledge,
-    });
-
-    await pledgeRef.ref.update(pledgeAttributes);
-
-    return pledgeAttributes;
+    // check current pledge here which will still be an object. always check currentPledge for pending, etc bc currentPledge is source of truth
+    const invitedAddress =
+      params.secondaryWalletAddress &&
+      currentPledge.wallets &&
+      currentPledge.wallets[params.secondaryWalletAddress] &&
+      currentPledge.wallets[params.secondaryWalletAddress].status === "pending";
+    const verifiedAddress =
+      params.secondaryWalletAddress &&
+      currentPledge.wallets &&
+      currentPledge.wallets[params.secondaryWalletAddress] &&
+      currentPledge.wallets[params.secondaryWalletAddress].status ===
+        "verified";
+
+    const snapshot = await admin
+      .firestore()
+      .collection("pledges")
+      .where(
+        `wallets.${params.secondaryWalletAddress}.status`,
+        "==",
+        "verified"
+      )
+      .get();
+    // if its already add it then un add from other pledge?
+    const isNotAlreadyAdded = snapshot.empty;
+    // check if secondary wallet is signing to accept
+    if (invitedAddress && isNotAlreadyAdded && params.secondaryWalletAddress) {
```

Comment:
All this branching logic is pretty hard to follow :(
It feels like this function is responsible many actions, is there a reason why we opted to add onto the existing endpoint instead of creating an each endpoint for handling adding and removing secondary wallets?